id/news/_posts/2019-12-12-separation-of-positional-and-keyword-arguments-in-ruby-3-0.md

```diff
@@ -0,0 +1,473 @@
+Anda ingin tetap menggunakan perilaku delegasi yang ditemukan pada Ruby 2.7
+dan sebelumnya, gunakan `ruby2_keywords`. Lihat bagian "Menangani delegasi
+argument" untuk lebih detail.
+
+## Kasus Khusus
+{: #typical-cases }
+
+Berikut adalah kasus yang paling khas. Anda dapat menggunakan operator *double
+splat* (`**`) untuk melewatkan *keyword* daripada sebuah *Hash*.
+
+{% highlight ruby %}
+# Method ini hanya menerima sebuah keyword argument
+def foo(k: 1)
+  p k
+end
+
+h = { k: 42 }
+
+# Pemanggilan method ini melewatkan sebuah positional Hash argument
+# Pada Ruby 2.7: Hash secara otomatis diubah menjadi sebuah keyword argument
+# Pada Ruby 3.0: Pemanggilan ini menyebabkan ArgumentError
+foo(h)
+  # => demo.rb:11: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
+  #    demo.rb:2: warning: The called method `foo' is defined here
+  #    42
+
+# Jika Anda ingin tetap menggunakan perilaku sebelumnya pada Ruby 3.0, gunakan double splat
+foo(**h) #=> 42
+{% endhighlight %}
+
+Ini adalah kasus lainnya. Anda dapat menggunakan tanda kurung kurawal (`{}`)
+untuk melewatkan sebuah objek *Hash* ketimbang *keyword* secara langsung.
+
+{% highlight ruby %}
+# Method ini menerima satu positional argument dan sebuah keyword rest argument
+def bar(h, **kwargs)
+  p h
+end
+
+# Pemanggilan ini hanya melewatkan sebuah keyword argument tanpa positional argument
+# Pada Ruby 2.7: keyword diubah ke sebuah positional Hash argument
+# Pada Ruby 3.0: Pemanggilan ini menyebabkan ArgumentError
+bar(k: 42)
+  # => demo2.rb:9: warning: Passing the keyword argument as the last hash parameter is deprecated
+  #    demo2.rb:2: warning: The called method `bar' is defined here
+  #    {:k=>42}
+
+# Jika Anda ingin tetap menggunakan perilaku sebelumnya pada Ruby 3.0, gunakan
+# tanda kurung kurawal untuk mengubahnya menjadi sebuah explicit Hash
+bar({ k: 42 }) # => {:k=>42}
+{% endhighlight %}
+
+## Apa yang sudah usang?
+{: #what-is-deprecated}
+
+Pada Ruby 2, *keyword argument* dapat dianggap sebagai *positional Hash argument*
+terakhir dan sebuah *positional Hash argument* terakhir dapat dianggap sebagai
+*keyword argument*.
+
+Karena pengubahan otomatis tersebut, kadang-kadang hal ini sangat kompleks dan
+sulit seperti yang dideskripsikan pada bagian terakhir. Sehingga, sekarang
+perilaku tersebut sudah usang pada Ruby 2.7 dan akan dihilangkan pada Ruby 3.
+Dengan kata lain, *keyword argument* akan dipisahkan seluruhnya dari
+*positional*-nya pada Ruby 3. Ketika Anda ingin melewatkan *keyword argument*,
+Anda seharusnya selalu menggunakan `foo(k: expr)` atau `foo(**expr)`. Jika
+Anda ingin menerima *keyword argument*, pada dasarnya Anda harus selalu
+menggunakan `def foo(k: default)` atau `def foo(k:)` atau `def foo(**kwargs)`.
+
+Ingat bahwa Ruby 3.0 tidak berperilaku berbeda ketika memanggil sebuah *method*
+yang tidak menerima *keyword argument* dengan *keyword argument*. Sebagai
```

Comment:
sorry this paragraph feels weird for me. how about this?

"Ingat bahwa Ruby 3.0 tidak membedakan perilaku ketika memanggil sebuah *method*
 yang tidak menerima *keyword argument* dengan *method* yang menerima *keyword argument*"

Reply:
After re-reading this paragraph, I felt the same way. I agree with your suggestion. Will update into it later.

Reply:
After rereading this paragraph, I felt the same way. I agree with your suggestion @kuntoaji. Will update into it later.

Reply:
Updated ya @kuntoaji